crates/compilers/src/compilers/resolc/compiler.rs

```diff
@@ -0,0 +1,196 @@
+            cmd.arg("--allow-paths");
+            cmd.arg(solc.allow_paths.iter().map(|p| p.display()).join(","));
+        }
+        if let Some(base_path) = &solc.base_path {
+            for path in solc.include_paths.iter().filter(|p| p.as_path() != base_path.as_path()) {
+                cmd.arg("--include-path").arg(path);
+            }
+
+            cmd.arg("--base-path").arg(base_path);
+            cmd.current_dir(base_path);
+        }
+
+        let child = if matches!(&input.language, SolcLanguage::Solidity) {
+            cmd.arg("--solc");
+            cmd.arg(&solc.solc);
+            cmd.arg("--standard-json");
+            let mut child = cmd.spawn().map_err(map_io_err(&self.resolc))?;
+            let mut stdin = io::BufWriter::new(child.stdin.take().unwrap());
+            serde_json::to_writer(&mut stdin, &input)?;
+            stdin.flush().map_err(map_io_err(&self.resolc))?;
+            child
+        } else {
+            cmd.arg("--yul");
+            cmd.arg(format!(
+                "{}",
+                &input
+                    .sources
+                    .first_key_value()
+                    .map(|k| k.0.to_string_lossy())
+                    .ok_or_else(|| SolcError::msg("No Yul sources available"))?
+            ));
+            cmd.arg("--bin");
+            cmd.spawn().map_err(map_io_err(&self.resolc))?
+        };
+
+        debug!("Spawned");
+
+        let output = child.wait_with_output().map_err(map_io_err(&self.resolc))?;
+        debug!("Finished compiling with standard json with status {:?}", output.status);
+
+        compile_output(output)
+    }
+
+    fn configure_cmd(&self, solc: &Solc) -> Command {
+        let mut cmd = Command::new(&self.resolc);
+        cmd.stdin(Stdio::piped()).stderr(Stdio::piped()).stdout(Stdio::piped());
+        cmd.args(&solc.extra_args);
+        cmd
+    }
+}
+
+fn map_io_err(resolc_path: &Path) -> impl FnOnce(std::io::Error) -> SolcError + '_ {
+    move |err| SolcError::io(err, resolc_path)
+}
+
+fn version_from_output(output: Output) -> Result<Version> {
+    if output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let version = stdout
+            .lines()
+            .filter(|l| !l.trim().is_empty())
+            .find(|l| l.contains("version"))
+            .ok_or_else(|| SolcError::msg("Version not found in resolc output"))?;
+
+        version
+            .split_whitespace()
+            .find(|s| s.starts_with("0.") || s.starts_with("v0."))
+            .and_then(|s| {
+                let trimmed = s.trim_start_matches('v').split('+').next().unwrap_or(s);
+                Version::from_str(trimmed).ok()
+            })
+            .ok_or_else(|| SolcError::msg("Unable to retrieve version from resolc output"))
+    } else {
+        Err(SolcError::solc_output(&output))
+    }
+}
+
+fn compile_output(output: Output) -> Result<Vec<u8>> {
+    // @TODO: Handle YUL output
```

Comment:
Is the YUL output different to solc?

Reply:
Ok, maybe we can add a better todo sentence to help others understand why we need to handle YUL output